### PR TITLE
Backup .plg files on updates

### PR DESCRIPTION
**Pull request by Squidly271**
To support rollbacks if required

### CODE REVIEW

**Comment by bergware** [comment]
The ability of doing a plugin rollback depends a lot on the plugin itself.

For example all Dynamix plugins maintain a single version. AKA the latest version. There is no way to downgrade once the new version is installed. This is done on purpose for several reasons.

Furthermore some of the plugin updates in the past were irreversible, e.g. convert to newer database format or relocation of files. These kind of actions make a rollback impossible.

In short a rollback possibility sounds interesting, but it will be limited in use.

**Comment by bergware** [comment]
Another strategy could be to let the plugin itself indicate whether a rollback is available (and which one).

This can be achieved by introducing a new attribute (e.g. **rollback="2016.09.22"**, but requires current plugins to be updated to support that.

At the same time wondering if all the effort is worthwhile...

**Comment by limetech** [comment]
Interesting idea but too much work to do it right.  Better to fix issues so that rollback is unnecessary.
